pkg/sdn/plugin/subnets.go

```diff
@@ -95,11 +95,6 @@ func (master *OsdnMaster) deleteNode(nodeName string) error {
 	if err != nil {
 		return fmt.Errorf("Error fetching subnet for node %q for deletion: %v", nodeName, err)
 	}
-	_, ipnet, err := net.ParseCIDR(sub.Subnet)
-	if err != nil {
-		return fmt.Errorf("Error parsing subnet %q for node %q for deletion: %v", sub.Subnet, nodeName, err)
-	}
-	master.subnetAllocator.ReleaseNetwork(ipnet)
```

Comment:
@pravisankar It will not break the normal case when node is deleted, because the hostsubnet is also deleted, and the watchSubnets in master will actually release the subnet

AddNode assigns the subnet because it is the place where hostsubnet is created. If we always end up creating hostsubnet without the 'subnet' field in it, then we will have to change the hostsubnet validator too. Something that we can consider later on, not necessary now.